### PR TITLE
Fix RxJS tests: ensure all assertions are called

### PR DESCRIPTION
**Pull request by natmegs**
Ensure all test assertions are being called correctly, even when relying on an async observable subscription.

This is an issue in unit tests using `rxjs-marbles` with non-marble tests that rely on `subscribe` calls.

## Testing

Unit tests pass.

## Trade-offs

N/A

## Dependencies

N/A

## Screenshots

N/A

### CODE REVIEW

File: plugins/jobs/src/js/data/__tests__/JobModel-test.js [hunk 1]
```diff
@@ -234,9 +234,9 @@ describe("JobModel Resolver", () => {
             jobsQuery({}, { path: [] }).pipe(take(1)),
             jobsQuery({}, { path: [] }).pipe(take(1)),
             jobsQuery({}, { path: [] }).pipe(take(1))
-          ).subscribe(jest.fn(), jest.fn(), () => {
-            expect(fetchJobs).toHaveBeenCalledTimes(1);
-          });
+          ).subscribe();
+          advance(20);
```
Comment (nLight) [inline plugins/jobs/src/js/data/__tests__/JobModel-test.js]: that's cool!

Comment (nLight) [reply plugins/jobs/src/js/data/__tests__/JobModel-test.js]: @DanielMSchmidt something we could try using in the data-layer extension plugin.

**Comment by natmegs** [comment]
@TattdCodeMonkey Good catch, not sure how I missed that 😳 Added another commit for that test but I think it uncovered an actual legit test failure (I `.skip`ed them in the commit). I do not see a mechanism for sharing observable subscriptions in the code... @nLight do you know if I'm missing something?

**Comment by TattdCodeMonkey** [comment]
Oh yes, I solved this when testing the new ui service data layer. Let’s sync up tomorrow and we can fix this.

**Comment by natmegs** [comment]
@TattdCodeMonkey I removed the faulty tests as discussed

**Pull request merged**